### PR TITLE
Setters validations

### PR DESCRIPTION
**Pull request by lmiotti**
### Summary
Added preconditions to check if User or Book is null, and if ID is positive for both cases.

### Trello Card
https://trello.com/c/khWxYGxs/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

**Review by rabinovichm — CHANGES_REQUESTED** [review]
The controllers should not contain any business logic. The validations should be done in the setters of each model. In order to make these validations check the models' restrictions specified on the Book model and User model trello cards.

File: training/src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -36,27 +42,47 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
...
         @ApiResponse(code = 404, message = "Book not found")
     })
+>>>>>>> master
     @ApiOperation(value = "Giving an id, returns a book", response = Book.class)
     public Book read(
         @ApiParam(value = "Id to find book", required = true) @PathVariable long id
     ) {
         return bookService.readBook(id);
 
+<<<<<<< HEAD
```
Comment (darkBasic666) [inline training/src/main/java/wolox/training/controllers/BookController.java]: marge conflicts

File: training/src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -1,9 +1,15 @@
 package wolox.training.controllers;
 
+<<<<<<< HEAD
```
Comment (leandrocostawolox) [inline training/src/main/java/wolox/training/controllers/BookController.java]: You commited this file without solve the conflict (There are some `<<<<< HEAD` and `>>>>>> master`

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]
Good! Just fix some merge conflict that you commited :)

**Review by leandrocostawolox — APPROVED** [review]
Good job!
Just to let you know, you can have a ErrorMessages class inside a utils package. This class could be final (to avoid inheritance) and have a static method which receives a String (called fieldName for example) and returns the custom messages, so you centralized there the message that mustn't be null

**Review by darkBasic666 — APPROVED** [review]

**Pull request merged**